cluster/pre-existing/util.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A library of helper functions for a pre-existing Kubernetes cluster
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/cluster/common.sh"
+source "${KUBE_ROOT}/cluster/lib/util.sh"
+
+function detect-project() {
+    if [[ -z "${MASTER_IP-}" ]]; then
+	echo "Set 'MASTER_IP' to the instance assigned to be the Kubernetes master" 1>&2
+	exit 1
+    fi
+
+    if [[ -z "${PROJECT-}" ]]; then
+	PROJECT="kubemark"
+    fi
+
+    if [[ -z "${SERVICE_CLUSTER_IP_RANGE-}" ]]; then
+	octets=($(echo "${MASTER_IP}" | sed -e 's|/.*||' -e 's/\./ /g'))
+	octets[3]=0
+	cluster_range=$(echo "${octets[*]}" | sed 's/ /./g')
+	SERVICE_CLUSTER_IP_RANGE="${SERVICE_CLUSTER_IP_RANGE:-$cluster_range}"
+    fi
+}
+
+function run-cmd-with-retries {
+  RETRIES="${RETRIES:-3}"
+  for attempt in $(seq 1 ${RETRIES}); do
+    local -r result=$("$@" 2>&1)
+    local -r ret_val="$?"
+    echo "${result}"
+    if [[ "${ret_val}" -ne "0" ]]; then
+      echo -e "${color_yellow}Attempt $attempt failed to $1 $2 $3. Retrying.${color_norm}" >& 2
+      sleep $(($attempt * 5))
+    else
+      echo -e "${color_green}Succeeded to $1 $2 $3.${color_norm}"
+      return 0
+    fi
+  done
+  echo -e "${color_red}Failed to $1 $2 $3.${color_norm}" >& 2
+  exit 1
+}
+
+function execute-cmd-on-master-with-retries() {
+  run-cmd-with-retries ssh kubernetes@"${MASTER_IP}" $@
```

Comment:
I guess we have `RETRIES="${2:-1}"` before the command (so we can allow setting RETRIES to a value different from 3). For eg. https://github.com/kubernetes/kubernetes/blob/master/test/kubemark/start-kubemark.sh#L91